Makefile

```diff
@@ -1,12 +1,21 @@
 #
-# github.com/docker/cli 
+# github.com/docker/cli
 #
 
+VERSION ?= unknown-version
+GITCOMMIT ?= $(shell git rev-parse --short HEAD 2> /dev/null || true)
+BUILDTIME ?= $(shell date --utc --rfc-3339 ns 2> /dev/null | sed -e 's/ /T/')
+
+LDFLAGS := -X github.com/docker/cli/cli.GitCommit=${GITCOMMIT} \
+	-X github.com/docker/cli/cli.BuildTime=${BUILDTIME} \
+	-X github.com/docker/cli/cli.Version=${VERSION} \
```

Comment:
I'd like to avoid adding a bunch of stuff to the Makefile that is really specific to just a single target (or a small number of targets).

Could we move this to a `script/build/ldflags` file (and add `export ...)?

Then I think we could `source script/build/ldflags` in the Makefile?

If that doesn't work maybe we can create `script/build/binary` and `script/build/cross` to source that file and call those from the Makefile?

I think the advantage to this approach is that it makes the dependency on these variables a lot more obvious.